connectors/src/connectors/slack_bot/index.ts

```diff
@@ -418,3 +436,72 @@ export class SlackBotConnectorManager extends BaseConnectorManager<SlackConfigur
     throw new Error("Method not implemented.");
   }
 }
+
+async function getFilteredChannels(
+  connectorId: number,
+  filterPermission: ConnectorPermission | null
+) {
+  const slackChannels: {
+    slackChannelId: string;
+    slackChannelName: string;
+    permission: ConnectorPermission;
+    private: boolean;
+  }[] = [];
+
+  if (
+    filterPermission &&
+    (filterPermission === "read" || filterPermission === "read_write")
+  ) {
+    // When requesting only read or read_write permissions, return empty array
+    return slackChannels;
+  }
+
+  const slackClient = await getSlackClient(connectorId, {
+    // Do not reject rate limited calls in update connector. Called from the API.
+    rejectRateLimitedCalls: false,
+  });
+
+  const [remoteChannels, localChannels] = await Promise.all([
+    getChannels(slackClient, connectorId, true),
+    SlackChannel.findAll({
+      where: {
+        connectorId,
+        // Here we do not filter out channels with skipReason because we need to know the ones that are skipped.
+      },
+    }),
+  ]);
+
+  const localChannelsById = localChannels.reduce(
+    (acc: Record<string, SlackChannel>, ch: SlackChannel) => {
+      acc[ch.slackChannelId] = ch;
+      return acc;
+    },
+    {}
+  );
+
+  for (const remoteChannel of remoteChannels) {
+    if (!remoteChannel.id || !remoteChannel.name) {
+      continue;
+    }
+
+    if (remoteChannel.is_private) {
+      // Skip private channels backend-side (displayed frontend-side if FF index_private_slack_channel is toggled)
```

Comment:
This comment is inaccurate no? if we don't list them here they won't be visible even with the FF, the FF does not really applies for `slack_bot` right?

Reply:
> if we don't list them here they won't be visible even with the FF

True

> This comment is inaccurate no?

The comment argues that we are skipping the channels on the backend-side. If we don't skip them here, they will be displayed conditionally based on the FF.

If not clear as-is, let me know what would make the comment clearer